src/Sarif.ValidationTests/DefaultObjectPopulatingVisitor.cs

```diff
@@ -0,0 +1,293 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.Json.Schema;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// This visitor will exhaustively populate a SARIF log with default values, starting at 
+    /// whatever level in the tree is visited. This has the effect of comprehehsively 
+    /// 'hydrating' all possible SARIF constructs as defined by the C# OM.
+    /// </summary>
+    public class DefaultObjectPopulatingVisitor : SarifRewritingVisitor
+    {
+        public delegate object PrimitiveValueBuilder(bool isRequired);
+
+        // The default builders create a placeholder value for all required property primitives
+        // and emits default (null, 0, false, etc.) values for the remainder. As a result, a
+        // default hydration of this class should contain no non-default values for 
+        // JSON primitive-valued properties that aren't required
+        public static IDictionary<Type, PrimitiveValueBuilder> PopulateRequiredPrimitives()
```

Comment:
>PopulateRequiredPrimitives [](start = 63, length = 26)

This method and the next don't do what their names say they do. Consider: `GetRequiredPrimitiveBuilders` (which isn't great because it suggests that it's the builders that are required), or perhaps better, `GetBuildersForRequiredPrimitives`.

For the other one, perhaps `GetBuildersForAllPrimitives`.